scripts/build/gen_relocate_app.py

```diff
@@ -435,7 +435,7 @@ def create_dict_wrt_mem():
 
         mem_region, copy_flag, file_name = line.split(':', 2)
 
-        file_name_list = glob.glob(file_name)
+        file_name_list = glob.glob(file_name, recursive=True)
```

Comment:
not opposed to the change itself, but do we have risk of side-effects ?

What happens in systems that might be building Zephyr inside Zephyr, or re-using source code.
Here a candidate could be `TF-M` which can build MCUboot as a secure bootloader.

Before approving this change, I would really like you to describe what problem you're solving, rather than just `It might be handy`.

Especially because the feature is not widely used, the risk is that side-effects are not discovered, as well as they could be hard to debug.

Should the proposed solution be configurable through an argument to this script ?
which then can be enable in Kconfig, similar to `CONFIG_CODE_DATA_RELOCATION`.

Reply:
I will look at the comments next week or the week after, once I'm back from the mountains...

Reply:
My main use case was this (the project runs on stm32h7x with plenty RAM free space and slow quadspi flash)

```
zephyr_code_relocate(${ZEPHYR_BASE}/subsys/**/*.c SRAM2_TEXT )
zephyr_code_relocate(${ZEPHYR_BASE}/drivers/**/*.c SRAM2_TEXT )
```

Relocating the zephyr system files lead to great speed up in my case. Specifying each subfolder in the example is definitely an option, but for me it increases complexity.

Original documentation of the `zephyr_code_relocate` macro is quite insufficient. When I read *limited regexp support* I had to go to the source and there I got the idea to add the recursive flag.

I agree it could possibly break some older code, but I have not found any usage now...